### PR TITLE
Container security methods

### PR DESCRIPTION
**Pull request by lsubash**
This patch captures documentation of container security,
keys involved and component update flow.

Signed-off-by: Subash Lakkimsetti <subash.lakkimsetti@intel.com>

### CODE REVIEW

File: source/security/container-security.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+Container Security
+------------------
+
+Container in |SPN| is a method for sub region support. Its an encapsulation of multiple components as depicted in the image below.
+
+.. image:: /images/sbl_container.PNG
+
+**Container**
+
+Container region header encapsulates details about its components and public key hashes or component hashes. Container header has to be signed using RSA private key. Respective public key hash would be store in |SPN| Hash Store and its verified while container gets registered.
```
Comment (mauricema) [inline source/security/container-security.rst:10]: "its verified" => "it is verified"

File: source/security/container-security.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+.. image:: /images/sbl_container.PNG
+
+**Container**
+
+Container region header encapsulates details about its components and public key hashes or component hashes. Container header has to be signed using RSA private key. Respective public key hash would be store in |SPN| Hash Store and its verified while container gets registered.
+
+**Components**
+
+Component region can be added as compressed form or raw images. These region supports both hash verification and siging with RSA private keys. Public key hash or component hash would be stored in container main header. These hashes would be verified when components gets loaded.
```
Comment (mauricema) [inline source/security/container-security.rst:14]: gets => get

File: source/security/container-security.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+**Mono Signing**
+
+Mono signing is a method in which all components region would be verified using RSA or with hash verification.
```
Comment (mauricema) [inline source/security/container-security.rst:18]: Mono signing is a method in which all components region would be signed in a single shot and verified using RSA or with hash verification.

File: source/security/container-security.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
++-----------------------------------+-------------------------------------------------------------------------------------+
+| OEM Defined                       | OEM to define and add to signing scripts                                            |
++-----------------------------------+-------------------------------------------------------------------------------------+
+| IPFW and others                   | KEY_ID_CONTAINER_RSA3072  (Container Def Key)                                       |
++-----------------------------------+-------------------------------------------------------------------------------------+
+
+
+**BOOT**
+
+This is for booting a linux kernel image. Refer :ref:`create-container-boot-image` for creation of bootable format.
```
Comment (mauricema) [inline source/security/container-security.rst:67]: linux kernel =>  OS kernel

**Comment by lsubash** [comment]
Fixed review comments.

File: source/security/container-security.rst [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+**KEYH**
+
+This is used external hash store creation. SBL maintains a loadable “Hash Store” to save digests of public keys used to sign external loadable by the bootloader.
+
+**OEM Defined**
+
+OEM can define their container methods.
+
+
+Container Firmware (IPFW)
```
Comment (mauricema) [inline source/security/container-security.rst:79]: It does not have to be "IPFW".

File: source/security/container-security.rst [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+**Key Requirement and Ownership**
+
+This picture depicts the various keys used for container components update on flash.
+
+.. image:: /images/ipfw_key_ownership.jpg
+   :align: center
+
+
+**IPFW Component Update Flow**
```
Comment (mauricema) [inline source/security/container-security.rst:94]: Container Component Update Flow

File: source/security/container-security.rst (hunk 5 above)
Comment (lsubash) [reply source/security/container-security.rst]: Updated.

File: source/security/container-security.rst (hunk 6 above)
Comment (lsubash) [reply source/security/container-security.rst]: Updated

**Review by rprangar — APPROVED** [review]

**Pull request merged**